spi/transfer-spi/src/test/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/DataRequestTest.java

```diff
@@ -32,4 +35,80 @@ void verifyNoDestination() {
         assertThrows(IllegalArgumentException.class, () -> DataRequest.Builder.newInstance().id(id).assetId(asset.getId()).build());
     }
 
+    @Test
+    void verifyCopy() {
+        DataRequest dataRequest = newSampleDataRequest();
+
+        var copy = dataRequest.copy();
+
+        assertSampleDataAddress(dataRequest, copy);
+    }
+
+    @Test
+    void verifyDeepCopy() {
+        DataRequest dataRequest = newSampleDataRequest();
+
+        var copy = dataRequest.copy();
+
+        var copyProperties = copy.getProperties();
+        copyProperties.put("foo", "new value");
```

Comment:
nit: use Faker to avoid magic strings, make correlation clearer when value used twice, and avoid accidental test success